src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
     }
-
-    // TODO: This section is commented because we do not have a factory of all the tokens. Once that is integrated then we will add the logic to get the tokens from the contract. For now we are getting the tokens from the token file
-    // async function initializeTokens() {
-    //   if (posManager !== null && provider !== null) {
-    //     // Variable to hold address of token0
-    //     let token0Addr = null
-
-    //     // Variable to hold address of token1
-    //     let token1Addr = null
-
-    //     // Variable to hold contract token0
-    //     let _token0: Contract
-
-    //     // Variable to hold contract token0
-    //     let _token1: Contract
-
-    //     // Variable to hold symbol of token0
-    //     let symbol0 = null
-
-    //     // Variable to hold symbol of token1
-    //     let symbol1 = null
-
-    //     token0Addr = await posManager.token0()
-    //     token1Addr = await posManager.token1()
-
-    //     if (token0Addr) {
-    //       _token0 = new ethers.Contract(token0Addr, IERC20Metadata.abi, accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider)
-    //       symbol0 = await _token0.symbol()
-    //       setToken0({address: token0Addr, symbol: symbol0, contract: _token0})
-    //     }
-
-    //     if (token1Addr) {
-    //       _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider)
-    //       symbol1 = await _token1.symbol()
-    //       setToken1({address: token1Addr, symbol: symbol1, contract: _token1})
-    //     }
-    //   }
-    // }
-
-    // initializeTokens()
     fetchData()
   }, [posManager])
 
   const withdraw = (amount: number) => {
+    let loading = notifyLoading('Waiting for withdraw reserves')
     withdrawLiquidity(amount)
       .then((res) => {
-        const { args } = res.events[1]
-        let message = "Add Liquidity Success: "
-          + args.pool
-          + args.reservesLen.toNumber()
-          + args.shares.toNumber()
-        notifySuccess(message)
+        notifyDismiss(loading)
+        // const { args } = res.events[1]
+        // let message = 'Withdraw Liquidity Success: ' + args.pool + args.reservesLen.toNumber() + args.shares.toNumber()
+        notifySuccess('Success')
       })
       .catch((err) => {
-        notifyError(err)
+        notifyDismiss(loading)
+        notifyError(err.toString())
       })
   }
 
+  const initializeGammaPoolContract = async () => {
+    if (provider && accountInfo && accountInfo?.address) {
+      if (!gammaPoolContract && GAMMA_POOL_ADDRESS) {
+        setGammaPoolContract(
+          new ethers.Contract(
+            GAMMA_POOL_ADDRESS,
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
+      }
+    } else {
+      notifyError('Please connect wallet')
+    }
+  }
```

Comment:
would these functions be better if we surround it with try and catch exceptions?

Reply:
Done